### DISCUSSION TITLE
fractionated gain ?

### DISCUSSION THREAD

**Discussion by QJesus** [fabricjs/fabric.js#7116] [Q&A]
a rect or circle, whatever, moving with cursor, Zoom in to display a local area of the canvas

something like below:

![image](https://user-images.githubusercontent.com/9971413/121636033-e08c8c80-cab9-11eb-9c4c-6f0d9e9fd16a.png)

**Comment by melchiar** [comment]
Please try to put some more effort in when writing your issue. There is an issue template for a reason.

Perhaps this will help
https://stackoverflow.com/a/23977061/7693185

**Comment by QJesus** [comment]
sorry it's not a bug, it's a feature or function.

something like the video shows, with the mouse moving, get a rect area from canvas directly, and gain it.

the question is: HOW to get the specified region from CANVAS directly? NOT object.clipTo


https://user-images.githubusercontent.com/9971413/121888719-e64cd100-cd4a-11eb-9d4b-35c22146d078.mp4

**Reply by asturur** [reply]
You can try to implement a brush that draws on the upper canvas and read this information from the area under the pointer.

**Reply by QJesus** [reply]
what's the api about 'read this information from the area' ?

**Reply by asturur** [reply]
there isn't a direct api for that.
You have to use either drawImage or getImageData/putImageData. You have to go native canvas.
